### ISSUE TITLE
Poor dialog and floating window hierarchy

### ISSUE THREAD

**Issue by artisteacher** [opentoonz/opentoonz#1846]
It's great no longer having the render dialog display on top of all open programs. However, when rendering directly from Output Settings, the render dialog displays underneath.
![screen shot 2018-03-06 at 8 10 02 am](https://user-images.githubusercontent.com/21159116/37034121-393723be-2116-11e8-80ee-12964b1a0af7.png)

There are occasionally other windows that unexpectedly display underneath something else when they are opened. It would be great if newly opened windows/dialogs always displayed on top of everything else - until the focus has changed.

**Comment by artisteacher** [comment]
Still happens on MacOS. I agree that the render dialog as it’s own window doesn’t have a great location either way but on top is more expected. 

 I really like how in Blender, the render progress bar (with an X to cancel) displays at the top of the screen & the progress bar also displays on the app icon in the dock.

**Comment by RodneyBaker** [comment]
Will tag as OS:Mac pending confirmation on other systems.

I hope to stare at the whole rendering flow more in the future.
I'd love to see the Farm Room become more of an all purpose Render room because many users likely see 'Farm' and say "I don't need no Farm' and they never explore that room again.
If it was a Render Room that also just happened to be able to render to network as well as local machine (which it can already) then users might be more inclined to explore it.  Then the Render Room itself could be plus'd up in useful ways.

And, because we can render behind the scenes with the Task/batch rendering, I hope someday we can gain an option to render to/through the batch renderer via the normal output settings dialog.  The benefit being we could launch our render in the background and immediately get back to work on the next thing.

**Comment by artisteacher** [comment]
Render dialog still displays beneath the output settings.

**Comment by artisteacher** [comment]
On a related note, the camera capture window displays on top of the preferences window, even when preferences is opened 2nd.

**Comment by RodneyBaker** [comment]
@artisteacher  Checking in to verify this issue is still valid and persisting.
It would be good to confirm if you are experiencing the issue in Tahoma2D as well.

**Comment by artisteacher** [comment]
Yes, it's definitely ongoing in both programs. Window focus is a little unpredictable in general. 

The render progress bar displays below the output settings window. Clicking on the render progress bar doesn't change the stacking order.

Some windows - ie. Output Settings, Flipbook or etc. always display above the program window. They might change stacking order with most other floating windows, but they never end up behind everything.

Preferences, Scene Settings, Camera Settings  or etc. end up hidden behind the program window if the program window gains focus.  You only notice they are there if you minimize the program window or move it to the side. Not super helpful...

**Comment by artisteacher** [comment]
On a related note, the Preferences window opens underneath a floating FX settings window. Windows shouldn't ever open underneath something else.

The vast majority of floating windows and dialogs shouldn't have different behaviors. What really should happen is this:

1. Newly opened floating windows or dialogs should always open on top of everything else
2. Clicking on a floating window/dialog should move it on top of everything else
3. Nothing should ever move behind the program window

**Comment by RodneyBaker** [comment]
@gab3d 
I started looking at this report but before spending too much time on it I'd appreciate if the problems here can be verified as still present.  I did open a draft fix to address the Preferences problem but I am mostly assuming the problem is still present.  I cannot verify as I have no monder MacOS system.

It may be that fix specifically targets the subject of this report.
If there is anything else outside of that I'll gladly look into it.

Thanks!

**Comment by RodneyBaker** [comment]
Closing as likely resolved but we need to conduct a thorough review of all things UI.
@artisteacher Please let us know if you are still experiencing the problem.
